packages/maskbook/src/plugins/BarnBridge/Model/SYPortfolioModel.tsx

```diff
@@ -0,0 +1,34 @@
+import { useAsyncRetry } from 'react-use'
+import { currentAccountSettings } from '../../Wallet/settings'
+
+export function SmartYieldPortfolioModelGetData() {
+    const protocolsMap: any = {}
+
+    return useAsyncRetry(async () => {
+        const walletAddress = currentAccountSettings.value
+        const response = await fetch(
+            `https://api.barnbridge.com/api/smartyield/users/${walletAddress}/portfolio-value`,
+            {
+                body: null,
+                method: 'GET',
+                mode: 'cors',
+                credentials: 'omit',
+            },
+        )
+
+        const realData = await response.json()
+
+        if (realData.data.length > 0) {
+            const data = realData.data[realData.data.length - 1]
+            return {
+                seniorValue: data.seniorValue,
+                juniorValue: data.juniorValue,
+            }
+        } else {
+            return {
+                seniorValue: 0,
+                juniorValue: 0,
+            }
+        }
```

Comment:
```suggestion
        import { last } from 'lodash-es'

        const data = last(realData.data)
        return {
            seniorValue: data?.seniorValue ?? 0,
            juniorValue: data?.juniorValue ?? 0,
        }
```

Reply:
ah much cleaner! thanks for the suggestion